eclipse/src/saros/ui/views/SarosView.java

```diff
@@ -113,38 +116,23 @@
 
   public static final String ID = "saros.ui.views.SarosView";
 
-  private final IRosterListener rosterListener =
-      new IRosterListener() {
-        /**
-         * Stores the most recent presence for each user, so we can keep track of away/available
-         * changes which should not update the RosterView.
-         */
-        private Map<String, Presence> lastPresenceMap = new HashMap<String, Presence>();
+  private final IContactsUpdate contactsUpdate =
+      new IContactsUpdate() {
+        private final Set<XMPPContact> wasAlreadyAvailable = new HashSet<>();
 
         @Override
-        public void presenceChanged(Presence presence) {
-
-          final boolean playAvailableSound =
-              preferenceStore.getBoolean(
-                  EclipsePreferenceConstants.SOUND_PLAY_EVENT_CONTACT_ONLINE);
-
-          final boolean playUnavailableSound =
-              preferenceStore.getBoolean(
-                  EclipsePreferenceConstants.SOUND_PLAY_EVENT_CONTACT_OFFLINE);
-
-          Presence lastPresence = lastPresenceMap.put(presence.getFrom(), presence);
-
-          if ((lastPresence == null || !lastPresence.isAvailable())
-              && presence.isAvailable()
-              && playAvailableSound) {
-            SoundPlayer.playSound(Sounds.USER_ONLINE);
-          }
-
-          if ((lastPresence != null)
-              && lastPresence.isAvailable()
-              && !presence.isAvailable()
-              && playUnavailableSound) {
-            SoundPlayer.playSound(Sounds.USER_OFFLINE);
+        public void update(Optional<XMPPContact> contactOptional, UpdateType type) {
+          if (type != IContactsUpdate.UpdateType.STATUS || !contactOptional.isPresent()) return;
+
+          XMPPContact contact = contactOptional.get();
```

Comment:
Please reformat this in some way. It is very hard to read without any blank lines at all.

Reply:
Yes, I rearranged some statements.